src/components/policy/policy_external/src/policy_manager_impl.cc

```diff
@@ -1774,12 +1774,15 @@ StatusNotifier PolicyManagerImpl::AddApplication(
   DeviceConsent device_consent = GetUserConsentForDevice(device_id);
   sync_primitives::AutoLock lock(apps_registration_lock_);
   if (IsNewApplication(application_id)) {
+    LOG4CXX_DEBUG(logger_, "Adding new application");
     AddNewApplication(application_id, device_consent);
     return std::make_shared<CallStatusChange>(update_status_manager_,
                                               device_consent);
   } else {
+    LOG4CXX_DEBUG(logger_, "Promote existed application");
     PromoteExistedApplication(application_id, device_consent);
-    return std::make_shared<utils::CallNothing>();
+    return std::make_shared<CallStatusChange>(update_status_manager_,
```

Comment:
@AByzhynar I would extract creating CallStatusChange from `if ` condition in case if we constructing it the same in both branches.